src/ralph/networks/admin.py

```diff
@@ -47,17 +47,64 @@ class Meta:
         exclude = ('parent',)
 
 
+def ip_address_base_object_link(obj):
+    if not obj.base_object:
+        return '&ndash;'
+    return '<a href="{}" target="_blank">{}</a>'.format(
+        reverse('admin:view_on_site', args=(
+            obj.base_object.content_type_id,
+            obj.base_object_id
+        )),
+        obj.base_object._str_with_type,
+    )
+
+
+class LinkedObjectTable(TableWithUrl):
+
+    def linked_object(self, item):
+        return ip_address_base_object_link(item)
+
+    linked_object.title = _('Linked object')
+
+
+class NetworkRalphChangeList(RalphChangeList):
+    def get_queryset(self, request):
+        """
+        Check if it has been used filter RangeFilterNetwork,
+        if yes change mptt_indent_field value in model_admin to '0'
+        so as not to generate a tree list
+
+        Args:
+            request: Django Request object
+
+        Returns:
+            Django queryset
+        """
+        queryset = super().get_queryset(request)
+        filters, *rest = self.get_filters(request)
+        range_filter = [
+            i for i in filters if isinstance(i, NetworkRangeFilter)
+        ]
+        if range_filter:
+            filter_params = self.get_filters_params()
+            if range_filter[0].lookup_kwarg in filter_params:
+                self.model_admin.mptt_indent_field = 10
+            else:
+                self.model_admin.mptt_indent_field = 0
```

Comment:
IMO it will not work in following case:
* user 1 visits networks admin (see mptt tree)
* user 1 applies range filter (see flat list)
* user 2 visits networks admin (see flat list, since mptt_indent_field is still 0 on model admin)